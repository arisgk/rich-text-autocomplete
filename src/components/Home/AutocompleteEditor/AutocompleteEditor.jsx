import React, { Component } from 'react';
import { CompositeDecorator, EditorState } from 'draft-js';
import Editor from 'draft-js-plugins-editor';
import 'draft-js-mention-plugin/lib/plugin.css';
import createMentionPlugin from 'draft-js-mention-plugin';
import createHashtagPlugin from './HashtagPlugin';

const mentionPlugin = createMentionPlugin();
const { MentionSuggestions } = mentionPlugin;

const hashtagPlugin = createHashtagPlugin();
const { Suggestions } = hashtagPlugin;

const plugins = [mentionPlugin, hashtagPlugin];

const styles = {
  root: {
    padding: 20,
    width: 600,
  },
  editor: {
    border: '1px solid #ddd',
    cursor: 'text',
    fontSize: 16,
    minHeight: 40,
    padding: 10,
  },
};

class AutocompleteEditor extends Component {
  constructor() {
    super();

    this.state = {
      editorState: EditorState.createEmpty(),
    };

    this.focus = () => this.refs.editor.focus();
    this.onChange = editorState => this.setState({ editorState });
    this.onSearchChange = this.onSearchChange.bind(this);
  }

  onSearchChange({ value }) {
    const { onMentionSearch } = this.props;
    onMentionSearch(value);
  }

  render() {
    const { mentionSuggestions } = this.props;

    return (
      <div style={styles.root}>
        <div style={styles.editor} onClick={this.focus}>
          <Editor
            editorState={this.state.editorState}
            onChange={this.onChange}
            ref="editor"
            plugins={plugins}
            spellCheck
          />
          <MentionSuggestions
            onSearchChange={this.onSearchChange}
            suggestions={mentionSuggestions}
          />
          <Suggestions
            onSearchChange={this.onSearchChange}
            suggestions={mentionSuggestions}
          />
        </div>
      </div>
    );
  }
}

export default AutocompleteEditor;
