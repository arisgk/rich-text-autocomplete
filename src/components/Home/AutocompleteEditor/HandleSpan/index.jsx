import React from 'react';

const styles = {
  container: {
    color: '#03A9F4',
    direction: 'ltr',
    unicodeBidi: 'bidi-override',
  },
};

const HandleSpan = ({ offsetKey, children }) => (
  <span
    style={styles.container}
    data-offset-key={offsetKey}
  >
    {children}
  </span>
);

export default HandleSpan;
