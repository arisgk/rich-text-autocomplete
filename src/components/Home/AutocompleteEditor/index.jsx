import React, { Component } from 'react';
import { CompositeDecorator, EditorState } from 'draft-js';
import Editor from 'draft-js-plugins-editor';
import 'draft-js-mention-plugin/lib/plugin.css';
import createMentionPlugin, { defaultSuggestionsFilter } from 'draft-js-mention-plugin';
import mentions from '../../../data/mentions';

const mentionPlugin = createMentionPlugin();
const { MentionSuggestions } = mentionPlugin;
const plugins = [mentionPlugin];

const styles = {
  root: {
    padding: 20,
    width: 600,
  },
  editor: {
    border: '1px solid #ddd',
    cursor: 'text',
    fontSize: 16,
    minHeight: 40,
    padding: 10,
  },
};

class AutocompleteEditor extends Component {
  constructor() {
    super();

    this.state = {
      editorState: EditorState.createEmpty(),
      suggestions: mentions,
    };

    this.focus = () => this.refs.editor.focus();
    this.onChange = editorState => this.setState({ editorState });
    this.onSearchChange = this.onSearchChange.bind(this);
  }

  onSearchChange({ value }) {
    this.setState({
      suggestions: defaultSuggestionsFilter(value, mentions),
    });
  }

  render() {
    return (
      <div style={styles.root}>
        <div style={styles.editor} onClick={this.focus}>
          <Editor
            editorState={this.state.editorState}
            onChange={this.onChange}
            ref="editor"
            plugins={plugins}
            spellCheck
          />
          <MentionSuggestions
            onSearchChange={this.onSearchChange}
            suggestions={this.state.suggestions}
            onAddMention={this.onAddMention}
          />
        </div>
      </div>
    );
  }
}

export default AutocompleteEditor;
