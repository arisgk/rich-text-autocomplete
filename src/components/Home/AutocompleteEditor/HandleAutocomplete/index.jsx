import React, { Component } from 'react';
import HandleSpan from './HandleSpan';
import SuggestionsPopover from './SuggestionsPopover';

const styles = {
  container: {
    position: 'relative',
  },
};

class HandleAutocomplete extends Component {
  constructor(props) {
    super(props);

    this.state = {
      open: false,
    };

    this.handleRequestClose = this.handleRequestClose.bind(this);
  }

  componentDidMount() {
    this.setState(prevState => ({
      ...prevState,
      open: true,
    }));
  }

  handleRequestClose() {
    this.setState(prevState => ({
      ...prevState,
      open: false,
    }));
  }

  render() {
    const { offsetKey, children } = this.props;
    console.log(this.props);

    return (
      <div
        data-offset-key={offsetKey}
        style={styles.container}
        ref={(rootDiv) => { this.rootDiv = rootDiv; }}
      >
        <HandleSpan>{children}</HandleSpan>
        <SuggestionsPopover
          open={this.state.open}
          anchorEl={this.rootDiv}
          handleRequestClose={this.handleRequestClose}
        />
      </div>
    );
  }
}

export default HandleAutocomplete;
