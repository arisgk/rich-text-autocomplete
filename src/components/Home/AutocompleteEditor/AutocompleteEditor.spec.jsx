/* global jest, describe, it, expect */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AutocompleteEditor from './AutocompleteEditor';

// -- https://github.com/facebook/draft-js/issues/702
jest.mock('draft-js/lib/generateRandomKey', () => () => '123');

describe('AutocompleteEditor', () => {
  it('renders correctly', () => {
    let mentionSuggestions = [];
    let hashtagSuggestions = [];
    let relationSuggestions = [];

    let wrapper = shallow(
      <AutocompleteEditor
        mentionSuggestions={mentionSuggestions}
        hashtagSuggestions={hashtagSuggestions}
        relationSuggestions={relationSuggestions}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();

    mentionSuggestions = [
      {
        name: 'Matthew Yorke',
        title: 'Senior Software Engineer',
        avatar: 'https://pbs.twimg.com/profile_images/517863945/mattsailing_400x400.jpg',
      },
      {
        name: 'Julian Smith',
        title: 'United Kingdom',
        avatar: 'https://avatars2.githubusercontent.com/u/1188186?v=3&s=400',
      },
    ];

    wrapper = shallow(
      <AutocompleteEditor
        mentionSuggestions={mentionSuggestions}
        hashtagSuggestions={hashtagSuggestions}
        relationSuggestions={relationSuggestions}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();

    mentionSuggestions = [];
    wrapper = shallow(
      <AutocompleteEditor
        mentionSuggestions={mentionSuggestions}
        hashtagSuggestions={hashtagSuggestions}
        relationSuggestions={relationSuggestions}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();

    hashtagSuggestions = [
      {
        name: 'nba',
      },
      {
        name: 'sweet',
      },
    ];

    wrapper = shallow(
      <AutocompleteEditor
        mentionSuggestions={mentionSuggestions}
        hashtagSuggestions={hashtagSuggestions}
        relationSuggestions={relationSuggestions}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();

    relationSuggestions = [
      {
        name: 'Meta',
      },
      {
        name: 'Oculus',
      },
    ];

    wrapper = shallow(
      <AutocompleteEditor
        mentionSuggestions={mentionSuggestions}
        hashtagSuggestions={hashtagSuggestions}
        relationSuggestions={relationSuggestions}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();

    hashtagSuggestions = [];
    relationSuggestions = [];
    wrapper = shallow(
      <AutocompleteEditor
        mentionSuggestions={mentionSuggestions}
        hashtagSuggestions={hashtagSuggestions}
        relationSuggestions={relationSuggestions}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
