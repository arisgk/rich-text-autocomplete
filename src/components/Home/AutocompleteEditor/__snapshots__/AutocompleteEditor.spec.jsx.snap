// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteEditor renders correctly 1`] = `
<div
  style={
    Object {
      "padding": 20,
      "width": 600,
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "border": "1px solid #ddd",
        "cursor": "text",
        "fontSize": 16,
        "minHeight": 40,
        "padding": 10,
      }
    }
  >
    <PluginEditor
      customStyleMap={Object {}}
      decorators={Array []}
      defaultBlockRenderMap={true}
      defaultKeyBindings={true}
      editorState={
        EditorState {
          "_immutable": Object {
            "allowUndo": true,
            "currentContent": Object {
              "blockMap": Object {
                "123": Object {
                  "characterList": Array [],
                  "data": Object {},
                  "depth": 0,
                  "key": "123",
                  "text": "",
                  "type": "unstyled",
                },
              },
              "entityMap": Object {
                "__add": [Function],
                "__create": [Function],
                "__get": [Function],
                "__getLastCreatedEntityKey": [Function],
                "__mergeData": [Function],
                "__replaceData": [Function],
                "add": [Function],
                "create": [Function],
                "get": [Function],
                "getLastCreatedEntityKey": [Function],
                "mergeData": [Function],
                "replaceData": [Function],
              },
              "selectionAfter": Object {
                "anchorKey": "123",
                "anchorOffset": 0,
                "focusKey": "123",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
              "selectionBefore": Object {
                "anchorKey": "123",
                "anchorOffset": 0,
                "focusKey": "123",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
            },
            "decorator": null,
            "directionMap": Object {
              "123": "LTR",
            },
            "forceSelection": false,
            "inCompositionMode": false,
            "inlineStyleOverride": null,
            "lastChangeType": null,
            "nativelyRenderedContent": null,
            "redoStack": Array [],
            "selection": Object {
              "anchorKey": "123",
              "anchorOffset": 0,
              "focusKey": "123",
              "focusOffset": 0,
              "hasFocus": false,
              "isBackward": false,
            },
            "treeMap": Object {
              "123": Array [
                Object {
                  "decoratorKey": null,
                  "end": 0,
                  "leaves": Array [
                    Object {
                      "end": 0,
                      "start": 0,
                    },
                  ],
                  "start": 0,
                },
              ],
            },
            "undoStack": Array [],
          },
        }
      }
      onChange={[Function]}
      plugins={
        Array [
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
        ]
      }
      spellCheck={true}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
  </div>
</div>
`;

exports[`AutocompleteEditor renders correctly 2`] = `
<div
  style={
    Object {
      "padding": 20,
      "width": 600,
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "border": "1px solid #ddd",
        "cursor": "text",
        "fontSize": 16,
        "minHeight": 40,
        "padding": 10,
      }
    }
  >
    <PluginEditor
      customStyleMap={Object {}}
      decorators={Array []}
      defaultBlockRenderMap={true}
      defaultKeyBindings={true}
      editorState={
        EditorState {
          "_immutable": Object {
            "allowUndo": true,
            "currentContent": Object {
              "blockMap": Object {
                "123": Object {
                  "characterList": Array [],
                  "data": Object {},
                  "depth": 0,
                  "key": "123",
                  "text": "",
                  "type": "unstyled",
                },
              },
              "entityMap": Object {
                "__add": [Function],
                "__create": [Function],
                "__get": [Function],
                "__getLastCreatedEntityKey": [Function],
                "__mergeData": [Function],
                "__replaceData": [Function],
                "add": [Function],
                "create": [Function],
                "get": [Function],
                "getLastCreatedEntityKey": [Function],
                "mergeData": [Function],
                "replaceData": [Function],
              },
              "selectionAfter": Object {
                "anchorKey": "123",
                "anchorOffset": 0,
                "focusKey": "123",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
              "selectionBefore": Object {
                "anchorKey": "123",
                "anchorOffset": 0,
                "focusKey": "123",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
            },
            "decorator": null,
            "directionMap": Object {
              "123": "LTR",
            },
            "forceSelection": false,
            "inCompositionMode": false,
            "inlineStyleOverride": null,
            "lastChangeType": null,
            "nativelyRenderedContent": null,
            "redoStack": Array [],
            "selection": Object {
              "anchorKey": "123",
              "anchorOffset": 0,
              "focusKey": "123",
              "focusOffset": 0,
              "hasFocus": false,
              "isBackward": false,
            },
            "treeMap": Object {
              "123": Array [
                Object {
                  "decoratorKey": null,
                  "end": 0,
                  "leaves": Array [
                    Object {
                      "end": 0,
                      "start": 0,
                    },
                  ],
                  "start": 0,
                },
              ],
            },
            "undoStack": Array [],
          },
        }
      }
      onChange={[Function]}
      plugins={
        Array [
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
        ]
      }
      spellCheck={true}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={
        Array [
          Object {
            "avatar": "https://pbs.twimg.com/profile_images/517863945/mattsailing_400x400.jpg",
            "name": "Matthew Yorke",
            "title": "Senior Software Engineer",
          },
          Object {
            "avatar": "https://avatars2.githubusercontent.com/u/1188186?v=3&s=400",
            "name": "Julian Smith",
            "title": "United Kingdom",
          },
        ]
      }
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
  </div>
</div>
`;

exports[`AutocompleteEditor renders correctly 3`] = `
<div
  style={
    Object {
      "padding": 20,
      "width": 600,
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "border": "1px solid #ddd",
        "cursor": "text",
        "fontSize": 16,
        "minHeight": 40,
        "padding": 10,
      }
    }
  >
    <PluginEditor
      customStyleMap={Object {}}
      decorators={Array []}
      defaultBlockRenderMap={true}
      defaultKeyBindings={true}
      editorState={
        EditorState {
          "_immutable": Object {
            "allowUndo": true,
            "currentContent": Object {
              "blockMap": Object {
                "123": Object {
                  "characterList": Array [],
                  "data": Object {},
                  "depth": 0,
                  "key": "123",
                  "text": "",
                  "type": "unstyled",
                },
              },
              "entityMap": Object {
                "__add": [Function],
                "__create": [Function],
                "__get": [Function],
                "__getLastCreatedEntityKey": [Function],
                "__mergeData": [Function],
                "__replaceData": [Function],
                "add": [Function],
                "create": [Function],
                "get": [Function],
                "getLastCreatedEntityKey": [Function],
                "mergeData": [Function],
                "replaceData": [Function],
              },
              "selectionAfter": Object {
                "anchorKey": "123",
                "anchorOffset": 0,
                "focusKey": "123",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
              "selectionBefore": Object {
                "anchorKey": "123",
                "anchorOffset": 0,
                "focusKey": "123",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
            },
            "decorator": null,
            "directionMap": Object {
              "123": "LTR",
            },
            "forceSelection": false,
            "inCompositionMode": false,
            "inlineStyleOverride": null,
            "lastChangeType": null,
            "nativelyRenderedContent": null,
            "redoStack": Array [],
            "selection": Object {
              "anchorKey": "123",
              "anchorOffset": 0,
              "focusKey": "123",
              "focusOffset": 0,
              "hasFocus": false,
              "isBackward": false,
            },
            "treeMap": Object {
              "123": Array [
                Object {
                  "decoratorKey": null,
                  "end": 0,
                  "leaves": Array [
                    Object {
                      "end": 0,
                      "start": 0,
                    },
                  ],
                  "start": 0,
                },
              ],
            },
            "undoStack": Array [],
          },
        }
      }
      onChange={[Function]}
      plugins={
        Array [
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
        ]
      }
      spellCheck={true}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
  </div>
</div>
`;

exports[`AutocompleteEditor renders correctly 4`] = `
<div
  style={
    Object {
      "padding": 20,
      "width": 600,
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "border": "1px solid #ddd",
        "cursor": "text",
        "fontSize": 16,
        "minHeight": 40,
        "padding": 10,
      }
    }
  >
    <PluginEditor
      customStyleMap={Object {}}
      decorators={Array []}
      defaultBlockRenderMap={true}
      defaultKeyBindings={true}
      editorState={
        EditorState {
          "_immutable": Object {
            "allowUndo": true,
            "currentContent": Object {
              "blockMap": Object {
                "123": Object {
                  "characterList": Array [],
                  "data": Object {},
                  "depth": 0,
                  "key": "123",
                  "text": "",
                  "type": "unstyled",
                },
              },
              "entityMap": Object {
                "__add": [Function],
                "__create": [Function],
                "__get": [Function],
                "__getLastCreatedEntityKey": [Function],
                "__mergeData": [Function],
                "__replaceData": [Function],
                "add": [Function],
                "create": [Function],
                "get": [Function],
                "getLastCreatedEntityKey": [Function],
                "mergeData": [Function],
                "replaceData": [Function],
              },
              "selectionAfter": Object {
                "anchorKey": "123",
                "anchorOffset": 0,
                "focusKey": "123",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
              "selectionBefore": Object {
                "anchorKey": "123",
                "anchorOffset": 0,
                "focusKey": "123",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
            },
            "decorator": null,
            "directionMap": Object {
              "123": "LTR",
            },
            "forceSelection": false,
            "inCompositionMode": false,
            "inlineStyleOverride": null,
            "lastChangeType": null,
            "nativelyRenderedContent": null,
            "redoStack": Array [],
            "selection": Object {
              "anchorKey": "123",
              "anchorOffset": 0,
              "focusKey": "123",
              "focusOffset": 0,
              "hasFocus": false,
              "isBackward": false,
            },
            "treeMap": Object {
              "123": Array [
                Object {
                  "decoratorKey": null,
                  "end": 0,
                  "leaves": Array [
                    Object {
                      "end": 0,
                      "start": 0,
                    },
                  ],
                  "start": 0,
                },
              ],
            },
            "undoStack": Array [],
          },
        }
      }
      onChange={[Function]}
      plugins={
        Array [
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
        ]
      }
      spellCheck={true}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={
        Array [
          Object {
            "name": "nba",
          },
          Object {
            "name": "sweet",
          },
        ]
      }
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
  </div>
</div>
`;

exports[`AutocompleteEditor renders correctly 5`] = `
<div
  style={
    Object {
      "padding": 20,
      "width": 600,
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "border": "1px solid #ddd",
        "cursor": "text",
        "fontSize": 16,
        "minHeight": 40,
        "padding": 10,
      }
    }
  >
    <PluginEditor
      customStyleMap={Object {}}
      decorators={Array []}
      defaultBlockRenderMap={true}
      defaultKeyBindings={true}
      editorState={
        EditorState {
          "_immutable": Object {
            "allowUndo": true,
            "currentContent": Object {
              "blockMap": Object {
                "123": Object {
                  "characterList": Array [],
                  "data": Object {},
                  "depth": 0,
                  "key": "123",
                  "text": "",
                  "type": "unstyled",
                },
              },
              "entityMap": Object {
                "__add": [Function],
                "__create": [Function],
                "__get": [Function],
                "__getLastCreatedEntityKey": [Function],
                "__mergeData": [Function],
                "__replaceData": [Function],
                "add": [Function],
                "create": [Function],
                "get": [Function],
                "getLastCreatedEntityKey": [Function],
                "mergeData": [Function],
                "replaceData": [Function],
              },
              "selectionAfter": Object {
                "anchorKey": "123",
                "anchorOffset": 0,
                "focusKey": "123",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
              "selectionBefore": Object {
                "anchorKey": "123",
                "anchorOffset": 0,
                "focusKey": "123",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
            },
            "decorator": null,
            "directionMap": Object {
              "123": "LTR",
            },
            "forceSelection": false,
            "inCompositionMode": false,
            "inlineStyleOverride": null,
            "lastChangeType": null,
            "nativelyRenderedContent": null,
            "redoStack": Array [],
            "selection": Object {
              "anchorKey": "123",
              "anchorOffset": 0,
              "focusKey": "123",
              "focusOffset": 0,
              "hasFocus": false,
              "isBackward": false,
            },
            "treeMap": Object {
              "123": Array [
                Object {
                  "decoratorKey": null,
                  "end": 0,
                  "leaves": Array [
                    Object {
                      "end": 0,
                      "start": 0,
                    },
                  ],
                  "start": 0,
                },
              ],
            },
            "undoStack": Array [],
          },
        }
      }
      onChange={[Function]}
      plugins={
        Array [
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
        ]
      }
      spellCheck={true}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={
        Array [
          Object {
            "name": "nba",
          },
          Object {
            "name": "sweet",
          },
        ]
      }
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={
        Array [
          Object {
            "name": "Meta",
          },
          Object {
            "name": "Oculus",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`AutocompleteEditor renders correctly 6`] = `
<div
  style={
    Object {
      "padding": 20,
      "width": 600,
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "border": "1px solid #ddd",
        "cursor": "text",
        "fontSize": 16,
        "minHeight": 40,
        "padding": 10,
      }
    }
  >
    <PluginEditor
      customStyleMap={Object {}}
      decorators={Array []}
      defaultBlockRenderMap={true}
      defaultKeyBindings={true}
      editorState={
        EditorState {
          "_immutable": Object {
            "allowUndo": true,
            "currentContent": Object {
              "blockMap": Object {
                "123": Object {
                  "characterList": Array [],
                  "data": Object {},
                  "depth": 0,
                  "key": "123",
                  "text": "",
                  "type": "unstyled",
                },
              },
              "entityMap": Object {
                "__add": [Function],
                "__create": [Function],
                "__get": [Function],
                "__getLastCreatedEntityKey": [Function],
                "__mergeData": [Function],
                "__replaceData": [Function],
                "add": [Function],
                "create": [Function],
                "get": [Function],
                "getLastCreatedEntityKey": [Function],
                "mergeData": [Function],
                "replaceData": [Function],
              },
              "selectionAfter": Object {
                "anchorKey": "123",
                "anchorOffset": 0,
                "focusKey": "123",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
              "selectionBefore": Object {
                "anchorKey": "123",
                "anchorOffset": 0,
                "focusKey": "123",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
            },
            "decorator": null,
            "directionMap": Object {
              "123": "LTR",
            },
            "forceSelection": false,
            "inCompositionMode": false,
            "inlineStyleOverride": null,
            "lastChangeType": null,
            "nativelyRenderedContent": null,
            "redoStack": Array [],
            "selection": Object {
              "anchorKey": "123",
              "anchorOffset": 0,
              "focusKey": "123",
              "focusOffset": 0,
              "hasFocus": false,
              "isBackward": false,
            },
            "treeMap": Object {
              "123": Array [
                Object {
                  "decoratorKey": null,
                  "end": 0,
                  "leaves": Array [
                    Object {
                      "end": 0,
                      "start": 0,
                    },
                  ],
                  "start": 0,
                },
              ],
            },
            "undoStack": Array [],
          },
        }
      }
      onChange={[Function]}
      plugins={
        Array [
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
          Object {
            "MentionSuggestions": [Function],
            "decorators": Array [
              Object {
                "component": [Function],
                "strategy": [Function],
              },
              Object {
                "component": [Function],
                "strategy": [Function],
              },
            ],
            "getAccessibilityProps": [Function],
            "handleReturn": [Function],
            "initialize": [Function],
            "onChange": [Function],
            "onDownArrow": [Function],
            "onEscape": [Function],
            "onTab": [Function],
            "onUpArrow": [Function],
          },
        ]
      }
      spellCheck={true}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
    <Decorated(Component)
      onSearchChange={[Function]}
      suggestions={Array []}
    />
  </div>
</div>
`;
